src/parameterizations/lateral/MOM_wave_drag.F90

```diff
@@ -105,23 +134,38 @@ subroutine wave_drag_calc(u, v, drag_u, drag_v, G, CS)
 
   is = G%isc ; ie = G%iec ; js = G%jsc ; je = G%jec
 
-  Drag_u(:,:) = 0.0 ; Drag_v(:,:) = 0.0
-
-  !$OMP do
-  do k=1,CS%nf ; do j=js,je ; do I=is-1,ie
-    Drag_u(I,j) = Drag_u(I,j) + u(I,j,k) * CS%coef_u(I,j,k)
-  enddo ; enddo ; enddo
-
-  !$OMP do
-  do k=1,CS%nf ; do J=js-1,je ; do i=is,ie
-    Drag_v(i,J) = Drag_v(i,J) + v(i,J,k) * CS%coef_v(i,J,k)
-  enddo ; enddo ; enddo
+  drag_u(:,:) = 0.0 ; drag_v(:,:) = 0.0
+
+  if (CS%tensor_drag) then
+    do k=1,CS%nf
+      call pass_vector(u(:,:,k), v(:,:,k), G%domain, direction=To_All+SCALAR_PAIR)
```

Comment:
Doing a separate blocking send for each of the `CS%nf` layers' worth of velocities will introduce unnecessary delays due to latency while waiting for the message passing to complete.  This would be much more efficient if the `do k` loop on line 140 were eliminated and line 141 were replaced with `call pass_vector(u(:,:,1:CS%nf), v(:,:,1:CS%nf), G%domain, direction=To_All+SCALAR_PAIR)`.